src/sentry/tasks/detect_invalid_group_status.py

```diff
@@ -0,0 +1,72 @@
+from collections import defaultdict
+from datetime import timedelta
+from logging import getLogger
+
+from django.utils import timezone
+
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistory, GroupHistoryStatus
+from sentry.silo import SiloMode
+from sentry.tasks.base import instrumented_task
+from sentry.types.group import SUBSTATUS_TO_STR, UNRESOLVED_SUBSTATUS_CHOICES, GroupSubStatus
+
+logger = getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.tasks.detect_invalid_group_status",
+    time_limit=65,
+    soft_time_limit=60,
+    silo_mode=SiloMode.REGION,
+)
+def detect_invalid_group_status(project_id: int):
+    # Find all unresolved groups for the organization
+    three_days_ago = timezone.now() - timedelta(days=3)
+    seven_days_ago = timezone.now() - timedelta(days=7)
+    groups = Group.objects.filter(
+        status=GroupStatus.UNRESOLVED,
+        project_id=project_id,
+    ).exclude(substatus=GroupSubStatus.ONGOING)
```

Comment:
Should this be paginated? At a glance it looks like it'd load an unbounded number of groups